### PR TITLE
Avoid `-Wincomplete-uni-patterns` in generated splices

### PR DESCRIPTION
**Pull request by sorki**
Resolves warnings coming from TH splices, like

```haskell
src/Ivory/Compile/C/Gen.hs:314:25: warning: [GHC-62161] [-Wincomplete-uni-patterns]
    Pattern match(es) are non-exhaustive
    In a pattern binding:
        Patterns of type ‘C.Type’ not matched:
            C.AntiType _ _
            C.Type (C.AntiDeclSpec _ _) _ _
            C.Type (C.AntiTypeDeclSpec _ _ _ _) _ _
    |
314 |       [C.BlockStm [cstm| for( $ty:(toType ty) $id:(toVar var)
```

### CODE REVIEW

File: Language/C/Quote/Base.hs [hunk 1]
```diff
@@ -224,19 +224,30 @@ qqIdE _                 = Nothing
...
+        case $(antiVarE v) of
+            C.Type (C.DeclSpec storage typeQuals typeSpec loc) _ _ ->
+                 C.DeclSpec
+                     (storage ++ $(dataToExpQ qqExp extraStorage))
+                     (typeQuals ++ $(dataToExpQ qqExp extraTypeQuals))
+                     typeSpec
+                     loc
+
+            x -> error
+                   $ "Impossible happened, expected C.Type but got "
```
Comment (sorki) [inline Language/C/Quote/Base.hs:237]: This one should maybe say expected `C.Type (C.DeclSpec {})`. Can improve iff needed / we don't figure out neater solution.

Comment (athas) [reply Language/C/Quote/Base.hs:237]: I'm OK with this.  It's an internal error, isn't it?

Comment (sorki) [reply Language/C/Quote/Base.hs]: It is and very unlikely to ever pop up I think. But I've made it nicer!

**Comment by athas** [comment]
You'll need to change those `<>`s to `++` for this to work on the various old GHCs that language-c-quote still supports.

**Comment by sorki** [comment]
> You'll need to change those `<>`s to `++` for this to work on the various old GHCs that language-c-quote still supports.

Ah, of course, that's why CI fails. Thanks!

**Pull request merged**